src/validation.cpp

```diff
@@ -1159,6 +1159,20 @@ CTransactionRef GetTransaction(const CBlockIndex* const block_index, const CTxMe
 {
     LOCK(cs_main);
 
+    if (mempool && !block_index) {
+        CTransactionRef ptx = mempool->get(hash);
+        if (ptx) return ptx;
+    }
+    if (g_txindex) {
+        CTransactionRef tx;
+        uint256 block_hash;
+        if (g_txindex->FindTx(hash, block_hash, tx)) {
+            // Transaction was found in a different block than requested.
+            if (block_index && block_index->GetBlockHash() != block_hash) return nullptr;
```

Comment:
Really edge case but I think this should be:
```cpp
if (!block_index || block_index->GetBlockHash() == block_hash) {
    hashBlock = block_hash;
    return tx;
}
```
So if the block hash doesn't match with the index then fall thru the next check - the transaction can be in another block right?

Reply:
indeed

Reply:
This shouldn't happen, though, as the `FindTx` wouldn't return true?

Reply:
txindex only keeps 1 block hash right?

Reply:
Oh, I didn't think of BIP 30

Reply:
Also, `block_index` could be an orphaned block where the transaction is but in the index the block is the one in the chain.